zinc-k8s-operator/src/zinc.py

```diff
@@ -0,0 +1,35 @@
+"""Standalone module for interacting with Zinc, with no charming context."""
+
+import logging
+import requests
+
+
+logger = logging.getLogger(__name__)
+
+# TODO: Add logging statements
+# TODO: Add tests
+
+
+class ZincAPI:
+    """Client for interacting with Zinc over HTTP."""
+    def __init__(self, port: int):
+        self.port = port
+        self.base_url = f"http://localhost:{self.port}"
+        # Define a requests session that retries failed connections.
+        self._session_with_retry = requests.Session()
+        adapter = requests.adapters.HTTPAdapter(
+            max_retries=requests.packages.urllib3.util.retry.Retry(
```

Comment:
Modern `requests` don't vendor `urrllib3`, thus:
```suggestion
            max_retries=urllib3.util.retry.Retry(
```